zipline/finance/trading.py

```diff
@@ -340,11 +341,25 @@ def previous_open_and_close(self, start_date):
     def next_market_minute(self, start):
         """
         Get the next market minute after @start. This is either the immediate
-        next minute, or the open of the next market day after start.
+        next minute, the open of the same day if start is before the market
+        open on a trading day, or the open of the next market day after start.
         """
         next_minute = start + datetime.timedelta(minutes=1)
         if self.is_market_hours(next_minute):
             return next_minute
+        # Check whether start is before the market open.
+        # If it is, then next_minute should be the market
+        # open.
+        if start.time() < time(hour=9, minute=31):
```

Comment:
Although will have to wrap the inequality check in a `try..else` to guard against `get_open_and_close()` being fed a `start` which isn't a trading day.